### ISSUE TITLE
webapp won't load - TypeError: src__common__types__provider

### ISSUE THREAD

**Issue by thor-jc** [flutter/flutter#119635]
## Steps to Reproduce

1. Upgraded from flutter 3.3.5 to 3.7.0
2. Launched web app to chrome

**Expected results:**  web app open an functional

**Actual results:**. web app hangs with javascript errors

<details>
<summary>js errors</summary>
<pre>
Uncaught SyntaxError: Unexpected token 'null'.                                                                     provider.dart.lib.js:15
</pre>

<pre>
program.dart.lib.js:24 Uncaught TypeError: Cannot read properties of undefined (reading 'src__common__types__provider')
    at load__packages__cel__src__cel__program_dart (program.dart.lib.js:24:71)
    at Object.execCb (require.js:1696:33)
    at Module.check (require.js:883:51)
    at Module.<anonymous> (require.js:1139:34)
    at require.js:134:23
    at require.js:1189:21
    at each (require.js:59:31)
    at Module.emit (require.js:1188:17)
    at Module.check (require.js:938:30)
    at Module.<anonymous> (require.js:1139:34)
    at require.js:134:23
    at require.js:1189:21
    at each (require.js:59:31)
    at Module.emit (require.js:1188:17)
    at Module.check (require.js:938:30)
    at Module.<anonymous> (require.js:1139:34)
    at require.js:134:23
    at require.js:1189:21
    at each (require.js:59:31)
    at Module.emit (require.js:1188:17)
    at Module.check (require.js:938:30)
    at Module.<anonymous> (require.js:1139:34)
    at require.js:134:23
    at require.js:1189:21
    at each (require.js:59:31)
    at Module.emit (require.js:1188:17)
    at Module.check (require.js:938:30)
    at Module.<anonymous> (require.js:1139:34)
    at require.js:134:23
    at require.js:1189:21
    at each (require.js:59:31)
    at Module.emit (require.js:1188:17)
    at Module.check (require.js:938:30)
    at Module.enable (require.js:1176:22)
    at Module.init (require.js:788:26)
    at callGetModule (require.js:1203:63)
    at Object.completeLoad (require.js:1590:21)
    at HTMLScriptElement.onScriptLoad (require.js:1717:29)
</pre>

![debug](https://user-images.githubusercontent.com/45845185/215820296-9f56a8a3-3a2f-4216-b4d9-138474a53a58.png)

![debug1](https://user-images.githubusercontent.com/45845185/215817667-41ca0711-b278-4bb3-92a7-4e82d21e8407.png)

![debug2](https://user-images.githubusercontent.com/45845185/215817827-a37b6dd1-2bdc-4beb-8efc-fcf5d8677688.png)

</details>

<details>
<summary>flutter doctor -v - WORKS</summary>

```

[✓] Flutter (Channel stable, 3.3.5, on macOS 13.1 22C65 darwin-x64, locale en-US)
    • Flutter version 3.3.5 on channel stable at /Users/thor/alloatech/env/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d9111f6402 (3 months ago), 2022-10-19 12:27:13 -0700
    • Engine revision 3ad69d7be3
    • Dart version 2.18.2
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/thor/Library/Android/sdk
    • Platform android-33, build-tools 33.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.74.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.58.0

[✓] Connected device (3 available)
    • LG H932 (mobile) • LGH932ba14dbad • android-arm64  • Android 9 (API 28)
    • macOS (desktop)  • macos          • darwin-x64     • macOS 13.1 22C65 darwin-x64
    • Chrome (web)     • chrome         • web-javascript • Google Chrome 109.0.5414.119

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```

</details>

<details>
<summary>flutter doctor -v - DOES NOT WORK</summary>

```

[✓] Flutter (Channel stable, 3.7.0, on macOS 13.1 22C65 darwin-x64, locale en-US)
    • Flutter version 3.7.0 on channel stable at /Users/thor/alloatech/env/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b06b8b2710 (8 days ago), 2023-01-23 16:55:55 -0800
    • Engine revision b24591ed32
    • Dart version 2.19.0
    • DevTools version 2.20.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/thor/Library/Android/sdk
    • Platform android-33, build-tools 33.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.74.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.58.0

[✓] Connected device (3 available)
    • LG H932 (mobile) • LGH932ba14dbad • android-arm64  • Android 9 (API 28)
    • macOS (desktop)  • macos          • darwin-x64     • macOS 13.1 22C65 darwin-x64
    • Chrome (web)     • chrome         • web-javascript • Google Chrome 109.0.5414.119

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```

</details>

**Comment by darshankawar** [comment]
Thanks for the report @thor-jc 
Can you provide a [complete minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) that shows the reported error ?

Also, looking at the error log, it seems to be similar to https://github.com/flutter/flutter/issues/102244 that you can check for your reference.

**Comment by thor-jc** [comment]
Thanks for the prompt response @darshankawar.   

#102244 sure does look the same with the same side effects - worked on other platform just not in web.  I wish my search would have found that one before I created this one.  My code base is pretty large and I have no idea what is causing it or I would have provided the code same, but it seems like that other one is reproducible.  

One comment:  I don't think it is "importing a file named null" is the root cause.  Rather it seems more like declaring a variable 'null' when it is probably a reserved word that has the compiler balking.   Looking forward to the fix as I am now stuck in 3.3.5 land.  

Regards,
John
